fastapi_mcp/prompts.py

```diff
@@ -0,0 +1,277 @@
+"""
+MCP Prompts support for FastAPI-MCP.
+
+This module provides FastAPI-native decorators and utilities for defining
+MCP-compliant prompts that can be discovered and executed by MCP clients.
+"""
+
+import logging
+from typing import Callable, List, Dict, Any, Optional, get_type_hints, Union
+from inspect import signature, Parameter, iscoroutinefunction
+import mcp.types as types
+
+from .types import PromptMessage, PromptArgument, TextContent
+
+logger = logging.getLogger(__name__)
+
+
+class PromptRegistry:
+    """Registry for managing MCP prompts in a FastAPI application."""
+
+    def __init__(self):
+        self.prompts: Dict[str, Dict[str, Any]] = {}
+
+    def register_prompt(
+        self, name: str, title: Optional[str] = None, description: Optional[str] = None, func: Optional[Callable] = None
+    ):
+        """
+        Register a prompt function with the registry.
+
+        Args:
+            name: Unique identifier for the prompt
+            title: Human-readable title for the prompt
+            description: Description of what the prompt does
+            func: The prompt function to register
+        """
+
+        def decorator(func: Callable) -> Callable:
+            # Extract argument schema from function signature
+            sig = signature(func)
+            type_hints = get_type_hints(func)
+
+            arguments = []
+            properties = {}
+            required = []
+
+            # Process function parameters to create prompt arguments
+            for param_name, param in sig.parameters.items():
+                if param_name in ["self", "cls"]:  # Skip self/cls parameters
+                    continue
+
+                param_type = type_hints.get(param_name, str)
+                is_required = param.default == Parameter.empty
+                param_desc = f"Parameter {param_name}"
+
+                # Try to extract description from docstring or annotations
+                if hasattr(param_type, "__doc__") and param_type.__doc__:
+                    param_desc = param_type.__doc__
+
+                arguments.append(PromptArgument(name=param_name, description=param_desc, required=is_required))
+
+                # Create JSON schema property
+                properties[param_name] = self._type_to_json_schema(param_type)
```

Comment:
let's just use pydantic here, if we pass data: (somepaynanticclass) it will automatically have funcitonallity to generate json schema and it will be well aligned to the fastapi and less magic